### DISCUSSION TITLE
Why widgets are on the top of every window?

### DISCUSSION THREAD

**Discussion by senkux** [elkowar/eww#583] [Q&A]
Isn't there any way to set widgets to the bottom of every window?

Also how to set widgets to be appear at only one particular workspace?

I'm on xfce

**Comment by oldwomanjosiah** [comment]
As for 1, if you are looking to have your widgets stack like a dock or toolbar (like polybar) and are using x11 you are going to want to set your `windowtype` to `dock`, or leave it unset while setting some `reserve`.

```ocaml
(defwindow bottom-bar
  :reserve (struts :distance "45px" :side "bottom")
  :geometry (geometry :anchor "bottom center")
  :vexpand "false"
  :stacking "fg"
  (child-widgets))
```

I don't currently use wayland, so I'm not sure exactly, but I think you can skip the `reserve` and just set `exclusive` to true instead.

- [x11](https://elkowar.github.io/eww/configuration.html#x11)
- [wayland](https://elkowar.github.io/eww/configuration.html#wayland)

As for 2, I don't believe that there is a wm-agnostic way to do this (on x at least) if it is at all possible. I am only able to test with i3 (on x11) at this time, but none of the window types keep the window on the same workspace.

**Comment by mantacid** [comment]
Regarding rendering widgets behind other windows: Try using `:wm-ignore true` and `:stacking “bg”`

As for the other question, you could try to make a script that detects what virtual desktop you’re on, then poll that in yuck & use it to set visibility.

**Reply by mantacid** [reply]
Also set the window type. That’s important too.

**Comment by elkowar** [comment]
On X, stacking windows into the background is more or less completely broken currently, as noone has been able to figure out _why_ it doesn't work. There's one or two issues open regarding this problem, too, follow those for more details
